js/api/BackgroundProcess.js

```diff
@@ -135,25 +133,25 @@ export function initializeGeolocationService() {
  */
 export function handleAppStateChange(state) {
   if (state === 'active') {
```

Comment:
I'd advise using ENUMs for the various 'states' as an Object.  There is even 3rd party ENUM classes if we'd want to take it one step further.  IMO, JavaScript ENUMs allow for a single Object to contain all possible states, which avoids 'ad-hoc' states mangled throughout the code.